### ISSUE TITLE
Aspect ratio, when using the deprecated option, does not work.

### ISSUE THREAD

**Issue by Duder-onomy** [imgix/ember-cli-imgix#73]
Currently, there is code that adds a `':1'`  to all aspect ratios passed in the deprecated format: https://github.com/imgix/ember-cli-imgix/blob/0a12f4ab65ae6ddfeb1b3866a46a43d8f8d87d99/addon/components/imgix-image.js#L186

When passing a valid aspectRatio of `4:3`, it adds a `:1` and then throws an invalid aspect ratio format error here: https://github.com/imgix/ember-cli-imgix/blob/0a12f4ab65ae6ddfeb1b3866a46a43d8f8d87d99/addon/components/imgix-image.js#L201

Subsequently, because it cannot parse that aspect ratio. The aspect ratio doe now work. 

Since this is a deprecated feature... maybe we can just ignore it.  
Passing the aspect ration the preferred way works great:
```
{{imgix-image
    path='/bingbong.jpg'
    options=(hash
        ar='4:3'
    )
}}
```

**Comment by Duder-onomy** [comment]
Now that I am trying to transition TheDyrt to using the official imgix lib, using srcsets and what not. 
I dont think the aspectRatio implementing in this lib works at all. I will do some digging

**Comment by frederickfogerty** [comment]
Hey @Duder-onomy! Although it may not be the best behavior, this is actually by design. I wanted to nudge people to move away from the old aspect ratio configuration, and so if we allowed people to pass the new format while still using the deprecated configuration option, it would be hard to have them move away from the deprecated usage and we would likely have to support the deprecated configuration longer. I hope this makes some sense!

So, when you say this "When passing a valid aspectRatio of 4:3" - I don't think this is a valid statement, as `4:3` has never been a valid format for aspectRatio.

Anyway, if we think it's better to support the new format with the old attribute, we can change this behavior of course.

> I dont think the aspectRatio implementing in this lib works at all.

This doesn't sound good at all! Please let us know if you find something that doesn't work.

**Comment by Duder-onomy** [comment]
Curious. In your mind what would be the correct way to express the 4:3 or 4x3 aspect ratio.  As long as I have been alive that is a 100% valid aspect ratio... as are most 2 numbers within reason.  If you mean to say that format is invalid.. I think some docs would suffice. Looking at the code, there is no possible way to pass a correct aspect ratio (using the deprecated pathway) as this prop will always have `:1` added to it. The code is clearly incorrect. Additionally, if you want someone to stop using a certain pattern you simply need to deprecate it. 

Lastly, Does imgix the service support aspect ratio at all? It seems that passing the prop on the path does nothing... and is no documentation for it on the imgix website. 

So yes. using the deprecated pathway AND the supported pathway do not work.

**Comment by Duder-onomy** [comment]
Either way. I have re-written your code in my fork and will open a PR once I get some testing on it. 
I am going to correctly deprecate the old pathway  of passing a `aspectRatio`  and encourage aspect ratio passed via an options obj. Until imgix the service supports the `ar` param, we will need to treat them the same.

**Comment by frederickfogerty** [comment]
Hey 👋 I'm definitely on the same page as you that `4:3` and `4x3` are valid aspect ratio formats, there's no debate about that! The thing is that this library, in the past, has only been able to handle decimal aspect ratios, and this is how it's been documented in previous versions (https://github.com/imgix/ember-cli-imgix/tree/4347765f4994ab314fb83dad1c36c2222d376a63#aspectratio). I based the upgrade to the new format off this behavior, and didn't expect people to start using the old parameter with the new format.

> The code is clearly incorrect.

I would appreciate some more constructive criticism than this - I have tried to outline my reasoning (and have elaborated on it above), so I'm not on the same page with this. We could probably implement a warning to prevent people from using the new format with the old parameter.

> Additionally, if you want someone to stop using a certain pattern you simply need to deprecate it.

The parameter is deprecated. https://github.com/imgix/ember-cli-imgix/blob/0a12f4ab65ae6ddfeb1b3866a46a43d8f8d87d99/addon/components/imgix-image.js#L153

> some docs would suffice.

As the feature is deprecated, it was removed from the documentation for the current versions. It was present in the documentation of old versions :)

> does imgix the service support aspect ratio at all?

This is something that @jayeb will have to respond to, but to my knowledge, it doesn't support it yet.

> using the deprecated pathway AND the supported pathway do not work.

You wrote in the first post in this issue that "passing the aspect ration the preferred way works great" - has this changed? There are extensive tests for both the deprecated and new parameter, so if there's an issue with the tests or something that we haven't covered, we'd love to hear about it!
Here are the tests: https://github.com/imgix/ember-cli-imgix/blob/0a12f4ab65ae6ddfeb1b3866a46a43d8f8d87d99/tests/integration/components/imgix-image-test.js#L57

> I have re-written the code in my fork and will open a PR once I get some testing on it. I am going to correctly deprecate the old pathway of passing a aspectRatio and encourage aspect ratio passed via an options obj. 

As discussed above, the parameter is already deprecated.

> Until imgix the service supports the ar param, we will need to treat them the same.

This will be up to you and @jayeb to decide if you both want to change the behavior.
